src/Adapter/Order/CommandHandler/AddOrderFromBackOfficeHandler.php

```diff
@@ -123,6 +127,30 @@ public function handle(AddOrderFromBackOfficeCommand $command)
         return new OrderId((int) $paymentModule->currentOrder);
     }
 
+    /**
+     * Saves customer message and link it to the cart.
+     *
+     * @param Cart $cart
+     * @param string $orderMessage
+     *
+     * @throws \PrestaShopDatabaseException
+     * @throws \PrestaShopException
+     */
+    private function addOrderMessage(Cart $cart, string $orderMessage): void
+    {
+        $messageId = null;
+        if ($oldMessage = Message::getMessageByCartId((int) $cart->id)) {
+            $messageId = $oldMessage['id_message'];
+        }
+        $message = new Message((int) $messageId);
+        if (Validate::isMessage($orderMessage)) {
```

Comment:
If the message is not valid then the save will be ignored silently, you should throw some `OrderConstraintException` here with a specific code related to INVALID_MESSAGE so that we can assign it a dedicated translated error message in the controller

Reply:
Done